ui/app/src/components/CreateSiteDialog.tsx

```diff
@@ -0,0 +1,616 @@
+/*
+ * Copyright (C) 2007-2019 Crafter Software Corporation. All Rights Reserved.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import React, { useEffect, useRef, useState } from 'react';
+import { withStyles } from '@material-ui/core/styles';
+import { get, post } from '../utils/ajax';
+import Dialog from "@material-ui/core/Dialog";
+import MuiDialogTitle from "@material-ui/core/DialogTitle";
+import Typography from '@material-ui/core/Typography';
+import DialogContent from '@material-ui/core/DialogContent';
+import IconButton from '@material-ui/core/IconButton';
+import CloseIcon from '@material-ui/icons/Close';
+import SearchIcon from '@material-ui/icons/Search';
+import Grid from '@material-ui/core/Grid';
+import Tab from '@material-ui/core/Tab';
+import Tabs from '@material-ui/core/Tabs';
+import BlueprintCard from './BlueprintCard';
+import Spinner from "./Spinner";
+import InputBase from '@material-ui/core/InputBase';
+import makeStyles from '@material-ui/core/styles/makeStyles';
+import SwipeableViews from 'react-swipeable-views';
+import Button from '@material-ui/core/Button';
+import clsx from 'clsx';
+import DialogActions from '@material-ui/core/DialogActions';
+import BluePrintForm from './BluePrintForm';
+import BluePrintReview from "./BluePrintReview";
+import CreateSiteLoading from "./CreateSiteLoading";
+import CreateSiteError from "./CreateSiteError";
+import { Blueprint } from '../models/Blueprint';
+import { Site, SiteState, Views } from '../models/Site';
+import { defineMessages, useIntl } from 'react-intl';
+import { Theme } from "@material-ui/core/styles/createMuiTheme";
+import CreateSiteDetails from "./CreateSiteDetails";
+import Empty from "./Empty";
+import { underscore } from '../utils/string';
+import updateToken from "../utils/token";
+
+const views: Views = {
```

Comment:
Is this Interface used anywhere else but here? Otherwise, might be worth putting it in here and not exporting it. Same for others of the same nature.

Reply:
No, I think is better keep it on different file to have more order, but if you want it on the same file so i will move it